src/v/cluster/cluster_link/tests/frontend_validation_test.cc

```diff
@@ -103,6 +104,18 @@ class frontend_validation_test : public seastar_test {
         co_return ec;
     }
 
+    ss::future<cluster::cluster_link::errc>
+    delete_mirror_topic(id_t id, delete_mirror_topic_cmd cmd) {
+        cluster::cluster_link_delete_mirror_topic_cmd del_cmd{id, cmd.copy()};
+        auto ec = _validator->validate_mutation(std::move(del_cmd));
+        if (ec == errc::success) {
+            auto err = co_await _table.local().apply_update(
+              testing::create_delete_mirror_topic_command(id, std::move(cmd)));
+            vassert(!err, "Failed to delete mirror topic: {}", err.message());
```

Comment:
I'd remove this assert, it is quite possible (atleast in theory) a mutation can fail validation on table update.